src/App.js

```diff
@@ -1,33 +1,45 @@
 import React, { Component } from 'react';
-import './App.scss';
 import { MoviesList } from './components/MoviesList';
+import { Search } from './components/Search';
 import moviesFromServer from './api/movies.json';
+import './App.scss';
 
 export class App extends Component {
-  state = {};
+  state = {
+    movies: moviesFromServer,
+    query: '',
+  };
+
+  getQuery = ({ target }) => {
+    this.setState({ query: target.value });
+  }
+
+  getFilteredMovies = () => {
+    const { movies, query } = this.state;
+
+    return [...movies].filter(movie => (
```

Comment:
```suggestion
    return movies.filter(movie => (
```
filter не мутирует исходный массив